addons/iap/models/iap.py

```diff
@@ -170,7 +170,16 @@ def get(self, service_name, force_create=True):
                 ('company_ids', 'in', self.env.companies.ids),
                 ('company_ids', '=', False)
         ]
-        accounts = self.search(domain, order='id desc')
+        accounts = self.search(domain + [('account_token', '!=', False)], order='id desc')
+        accounts_without_token = self.search(domain + [('account_token', '=', False)])
```

Comment:
instead of doing 2 searches, you could filter found accounts by account_token value.

Reply:
You are right, I think it is better this way, what do you think ?